crates/levm/src/operations.rs

```diff
@@ -163,17 +163,41 @@ impl Operation {
             // Operation::Tload => Bytes::copy_from_slice(&[Opcode::TLOAD as u8]),
             // Operation::Tstore => Bytes::copy_from_slice(&[Opcode::TSTORE as u8]),
             Operation::Mcopy => Bytes::copy_from_slice(&[Opcode::MCOPY as u8]),
-            // Operation::Push0 => Bytes::copy_from_slice(&[Opcode::PUSH0 as u8]),
-            // Operation::Push((n, x)) => {},
+            Operation::Push0 => Bytes::copy_from_slice(&[Opcode::PUSH0 as u8]),
+            Operation::Push((n, value)) => {
+                assert!(*n <= 32, "PUSH32 is the max");
+                // the amount of bytes needed to represent the value must
+                // be less than the n in PUSHn
+                assert!(
+                    value.bits().div_ceil(8) <= *n as usize,
+                    "value can't doesn't fit in n bytes"
+                );
+                let mut word_buffer = [0; 32];
+                value.to_big_endian(&mut word_buffer);
+                // extract the last n bytes to push
+                let value_to_push = &word_buffer[((32 - *n) as usize)..];
+                assert_eq!(value_to_push.len(), *n as usize);
+                let opcode = Opcode::from(Opcode::PUSH0 as u8 + *n);
+                let mut bytes = vec![opcode as u8];
+                bytes.extend_from_slice(value_to_push);
+
+                Bytes::copy_from_slice(&bytes)
+            }
             Operation::Push32(value) => {
```

Comment:
Shouldn't we delete this `Push32` if we have the `Push((n, value))`?

Reply:
`Push32` was already defined separately from `Pushn` and `Push0`, so I didn't touch that. It may seem redundant, but maybe it's a feature and not a bug. If it becomes a problem, we can open another issue to solve the redundancy. Is that okay with you?

```rust
pub enum Operation {
    ...
    Push0,
    Push32(U256),
    Push((u8, U256)),
    ...
}
```

Reply:
Okay, also `Push32` is one of the Push ops that we use the most, it's fine then